src/components/AddressSummary.js

```diff
@@ -0,0 +1,199 @@
+   * quantity {number} Number of transactions in this address
+   * loading {boolean} If is waiting server response
+   * errorMessage {String} Message to be shown in case of an error
+   * selectedToken {String} UID of the selected token when address has many
+   */
+  state = {
+    balance: {},
+    quantity: null,
+    loading: true,
+    errorMessage: '',
+    selectedToken: '',
+  }
+
+  componentDidMount() {
+    this.getData();
+
+    WebSocketHandler.on('network', this.handleWebsocket);
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.address !== prevProps.address) {
+      // Address changed, must update date
+      this.getData();
+    }
+  }
+
+  componentWillUnmount() {
+    WebSocketHandler.removeListener('network', this.handleWebsocket);
+  }
+
+  /**
+   * Called when 'network' ws message arrives
+   * If it's a new tx message update data, in case is necessary
+   *
+   * @param {Object} wsData Data from websocket
+   */
+  handleWebsocket = (wsData) => {
+    if (wsData.type === 'network:new_tx_accepted') {
+      if (this.props.shouldUpdate(wsData)) {
+        this.getData();
+      }
+    }
+  }
+
+  /**
+   * Request data from server and update state balance and quantity
+   */
+  getData() {
+    hathorLib.walletApi.getAddressBalance(this.props.address, (response) => {
+      if (response.success) {
+        let selectedToken = '';
+        if (this.state.selectedToken) {
+          // If user had selected a token already, should continue the same
+          selectedToken = this.state.selectedToken;
+        } else {
+          const hathorUID = hathorLib.constants.HATHOR_TOKEN_CONFIG.uid
+          if (hathorUID in response.tokens_data) {
+            // If HTR is in the token list of this address, it's the default selection
+            selectedToken = hathorUID;
+          } else {
+            // Otherwise we get the first element
+            const keys = Object.keys(response.tokens_data);
+            selectedToken = response.tokens_data[keys[0]];
+          }
+        }
+
+        this.setState({
+          balance: response.tokens_data,
+          quantity: response.quantity,
+          loading: false,
+          selectedToken,
+        });
+      } else {
+        this.setState({
+          loading: false,
+          errorMessage: response.message,
+        });
+      }
+    });
+  }
+
+  /**
+   * Called when selected token is changed
+   *
+   * @param {Object} e Event emitted when select is changed
+   */
+  selectChanged = (e) => {
+    this.setState({ selectedToken: e.target.value });
+  }
+
+  render() {
+    if (Object.keys(this.state.balance).length === 0) {
+      return null;
+    }
+
+    const loadMainInfo = () => {
+      return (
+        <div className="card text-white bg-dark mb-3">
+          <div className="card-body">
+            Address: {this.props.address}<br />
```

Comment:
Missing i18n markers.

Reply:
We don't have i18n in the explorer, should we do something?